### PR TITLE
Add @ shortcut to open project selection when typing TE description

### LINKED ISSUE #4436: Add @ shortcut to open project selection when typing TE description
### 💻 Environment
Platform: macOS

### 📒 Description
Allow users to use `@` shortcut in the description field to search and chose a project.

[Miro link](https://miro.com/app/board/o9J_kskXHPw=/?moveToWidget=3074457348487714830&cot=3)

### PR DESCRIPTION
**Pull request by AndrewVebster**
### 📒 Description
This PR allows users to use `@` shortcut in the description field to search and chose a project.

### 🕶️ Types of changes
- **New feature** (non-breaking change which adds functionality)
- **Breaking change** (fix or feature that would cause existing functionality to change) - can break autocomplete

### 🤯 List of changes
- `TimerViewController` now is not a delegate for `descriptionTextField`
- `TimerDescriptionFieldHandler` contains the logic with shortcuts detection and autocomplete filtering
- `TimerDescriptionFieldHandler` also is partially covered with unit tests

### 👫 Relationships
Closes #4436 

### 🔎 Review hints
Project shortcut feature is disabled by default. To enable it:
Go to `TimerDescriptionFieldHandler.swift` -> `line 61` -> set `enableProjectShortcut` to `true`

### CODE REVIEW

**Comment by AndrewVebster** [comment]
Linter warning is fixed in #4496

File: src/ui/osx/TogglDesktop/Common/Extensions/String+Search.swift [hunk 1]
```diff
@@ -32,4 +32,47 @@ extension String {
...
+            let query = String(queryWithToken.dropFirst())
+
+            return (token, query)
+
+        } catch {
+            return (nil, "")
+        }
+    }
+
+    static func matches(for regex: String, in text: String) -> [String] {
```
Comment (skel35) [inline src/ui/osx/TogglDesktop/Common/Extensions/String+Search.swift:65]: unused method?

**Review by skel35 — APPROVED** [review]
Great job!
I really like how the shortcut is working!

Kudos for writing the first tests of the macOS app 😄

File: src/ui/osx/TogglDesktop/Common/Extensions/String+Search.swift (hunk 1 above)
Comment (AndrewVebster) [reply src/ui/osx/TogglDesktop/Common/Extensions/String+Search.swift]: good catch, removed, thanks

**Pull request merged**